### ISSUE TITLE
Error when using SAS-policy and connection string with Service Bus Queue

### ISSUE THREAD

**Issue by AntonBarwald** [paolosalvatori/ServiceBusExplorer#611]
When I use **RootManageSharedAccessKey** for the Service Bus Namespace I can see everything, see below
![image](https://user-images.githubusercontent.com/10939483/147639493-680f0296-190b-4c3b-80e2-bc43beb315fd.png)

When creating a SAS policy (Manage, Send, Listen) on the queue only and connecting using the connection string:

- `<08:55:02> Failed to retrieve Service Bus topics. Exception: Microsoft.ServiceBus.Messaging.MessagingException: Cannot get entity 'my-queue' because it is not of type TopicDescription. Check that you are using method(s) with the correct entity type. ---> System.Runtime.Serialization.SerializationException: Error in line 1 position 2599. Expecting element 'TopicDescription' from namespace 'http://schemas.microsoft.com/netservices/2010/10/servicebus/connect'.. Encountered 'None'  with name '', namespace ''. `

- `<Error><Code>401</Code><Detail>User is not authorized. TrackingId:...-...</Detail></Error>`

![image](https://user-images.githubusercontent.com/10939483/147640811-7754661f-a86f-4d0f-b2cc-0898b77579ea.png)

When creating a SAS policy (Manage, Send, Listen) on the topic and working with subscriptions it seems to work:

![image](https://user-images.githubusercontent.com/10939483/147640684-9ca34c1b-0839-42a7-b886-79a49e129bb0.png)

Anything that I might have missed when connecting to queues?

 Thanks for assisting!

**Comment by ErikMogensen** [comment]
For me it does not work with queues nor topics. There are at least two PRs that added/enhanced this funtionality, #375 and #562. 

@rajrao and @avdv, does it work for you?

**Comment by chandraatgithub** [comment]
Latest version 5.0.7 also has the same issue "User is not authorized" while using SAS-Policy in Queue, whereas only Topic Subscription works fine.

**Comment by cosminstirbu** [comment]
Any news on this issue. We're facing the same problem.

**Comment by cosminstirbu** [comment]
Based on my local debugging, it seems the issue is caused by the usage of [ServiceBusHelper2](https://github.com/paolosalvatori/ServiceBusExplorer/blob/develop/src/ServiceBus/Helpers/ServiceBusHelper2.cs) in [HandleQueueControl](https://github.com/paolosalvatori/ServiceBusExplorer/blob/develop/src/ServiceBusExplorer/Controls/HandleQueueControl.cs#L302) and [HandleTopicControl](https://github.com/paolosalvatori/ServiceBusExplorer/blob/develop/src/ServiceBusExplorer/Controls/HandleTopicControl.cs#L157)

Basically when you select a queue / topic with a entity level policy it attempts to use the `ServiceBusAdministrationClient` (via `ServiceBusHelper2`) (and it fails, as expected).

On my local machine, to simply work around this, I've commented the usage of `ServiceBusHelper2` in `HandleQueueControl`, however I would defer to you for a long term solution, as I'm not familiar enough with the original intentions behind this design.

**Comment by ErikMogensen** [comment]
When testing this on the latest commit in the develop branch both connecting to a queue and a topic works for me but I do get tons of error messages in the log window. 

Is it only the error messages that is the problem?

**Comment by cosminstirbu** [comment]
It connects successfully, however if you select the topic/queue the view to the right of the tree view doesn't load, and an exception is logged.

**Comment by cosminstirbu** [comment]
@ErikMogensen - not sure if you got the chance to look into this.

**Comment by ErikMogensen** [comment]
@AntonBarwald, @chandraatgithub, @cosminstirbu 
Please test https://github.com/paolosalvatori/ServiceBusExplorer/releases/tag/5.0.15-preview and report back whether this issue is resolved or not. Please keep an eye out for issues regarding connecting.

**Comment by cosminstirbu** [comment]
Tested the preview version and it fixed the issue. So far so good, didn't find any other issues, although my testing was rather limited.

Thank you!

**Comment by ErikMogensen** [comment]
Happy to read that!
